pkg/controller/vitesskeyspace/reconcile_resharding.go

```diff
@@ -136,3 +156,33 @@ func (r *reconcileHandler) reconcileResharding(ctx context.Context) (reconcile.R
 
 	return resultBuilder.Result()
 }
+
+func (r reconcileHandler) shardsRowCount(ctx context.Context, shardNames []string) (uint64, error) {
+	var rowCount uint64
+	for _, shardName := range shardNames {
+		tabletMap, err := r.ts.GetTabletMapForShard(ctx, r.vtk.Spec.Name, shardName)
```

Comment:
Since we're only interested in the master, we could just call GetShard and find the master_alias there. That way we don't have to fetch all the non-master tablet records.

Reply:
Ahhh, good catch. Will update.

Reply:
Updated.